src/timestamp.rs

```diff
@@ -187,6 +187,46 @@ impl Timestamp {
     }
 }
 
+#[cfg(all(
+    feature = "std",
+    not(miri),
+    any(
```

Comment:
I think we shouldn't need to `cfg` out these impls on `wasm32-unknown-unknown`. The value of `UNIX_EPOCH` on `wasm32-unknown-unknown` is meaningless, but you also can't construct a `SystemTime` anyways, so I think leaving these impls is ok, and will be less likely to result in impossible build failures for anyone targeting multiple platforms.